robottelo/ui/base.py

```diff
@@ -539,11 +540,14 @@ def click(self, locator, wait_for_ajax=True,
         :raise: UINoSuchElementError if the element could not be found.
 
         """
-        element = self.wait_until_element(locator, timeout=waiter_timeout)
+        if type(locator).__name__ == 'tuple':
+            element = self.wait_until_element(locator, timeout=waiter_timeout)
+        else:
+            element = locator
         if element is None:
             raise UINoSuchElementError(
-                '{0}: element with locator {1} not found while trying to click'
-                .format(type(self).__name__, locator)
+                '{0}: element was not found while trying to click'
```

Comment:
I feel that I give bad explanation. Let's me show few scenarios that our new method should handle:
1) Classical with locators:
`self.click(tab_locators['ak.host_collections'])` (we can show locator information in our error message)
2) With element:
```
activation_key = self.search(name)
self.click(activation_key)
```
Here we can show a lot of nice information in our error
3) None:
```
activation_key = self.search(name) 
-> Return None
self.click(activation_key)
```
Here we have no information

All that logic needs to be inside of single method and situation from case 3) too as we should prevent bunch of redundant ifs